### PR TITLE
make LIVE_UPDATE=1 live

### PR DESCRIPTION
**Pull request by rouming**
Another take on speeding the `make live` up. This based on the previous attempt: #3842 and latest changes in the linuxkit (thanks, @deitch), which is able to generate the tarball much faster (see the corresponding LIVE_FAST=1 option and the 3rd d887917c6c33e2e9f0e6 commit). The idea has been slightly changed: the qcow2 live image will be updated with the whole rootfs tarball, and not only the tarball which contains the specified package. Yes, we lose in numbers (10s vs 30s), but should be less error prone: no need to specify the exact package, no yq magic. Hopefully this is only the first step and we won't stop and continue build system improvements.

To revieweres: all comments from the #3842 should be addressed here.

Performance-wise we have:

###### Regular build, squashfs
```
$ make ROOTFS_FORMAT=squashfs live

real    3m40.847s
user    1m15.398s
sys     0m5.805s

```
###### Regular build, ext4
```
$ make ROOTFS_FORMAT=ext4 live

real    2m10.059s
user    1m46.358s
sys     0m6.703s
```
###### No live.qcow2 was previously generated, one-time operation
```
$ make LIVE_UPDATE=1 live

real    1m3.610s
user    0m55.851s
sys     0m5.893s

```

###### No changes of rootfs tarball, so live.qcow2 was not touched at all
```
$ make LIVE_UPDATE=1 live

real    0m20.289s
user    0m12.689s
sys     0m4.753s
```

###### pkg/pillar was modified
```
$ make LIVE_UPDATE=1 live

real    0m26.959s
user    0m21.706s
sys     0m10.710s
```

So basically this should give us ~30s of the whole live image generation for each development cycle.

Limitations and differences: 
1. EXT4 as rootfs, which bloats the image from ~200Mb to ~800Mb
2. During the live update only rootfs will be updated, keeping the /config partition same, which means onboarded node stays onboarded, which I consider as a huge win!

CC: @deitch

### CODE REVIEW

File: docs/BUILD.md [hunk 1]
```diff
@@ -377,6 +377,22 @@ The current process uses the control knob `PLATFORM` in the following places in
 * As the first argument to [prepare-platform.sh](../tools/prepare-platform.sh) in the Makefile targets for `live.*`, `installer.*` and `verification.*`, which, in turn, uses it to add specific files to the build and output directories prior to making the final image.
 * In the final eve image `lf-edge/eve`, specifically the entrypoint [runme.sh](../pkg/eve/runme.sh), where the platform is passed as `-p` argument when calling `docker run`, and is used to modify the final layout.
 
+### Note for developers
+
+`make live` typically requires a significant amount of time, approximately 3 minutes, to generate a tarball and the resulting squashfs image. This delay can be frustrating during daily development cycles of code compilation, live testing, and iteration. For developers who are less patient, optimizing the live image build process is essential. This optimization entails a few adjustments to the build procedure:
```
Comment (deitch) [inline docs/BUILD.md:382]: The right idea, but worth making the language a little "nicer". How about something like:

"This delay can significantly slow down the development cycles of code compilation, live testing, and iteration. During development, we support rapid live image update, for use during development only. This optimization uses the following procedure:"

File: docs/BUILD.md [hunk 2]
```diff
@@ -377,6 +377,22 @@ The current process uses the control knob `PLATFORM` in the following places in
...
+
+1. Creating a live image with an EXT4 root filesystem. The live command also supports the EXT4 format for the root filesystem, resulting in a larger image size. However, this format allows for a mutable root filesystem, enabling it to be remounted with read-write (RW) options. This flexibility provides greater freedom during EVE runtime as well as during the live image creation, facilitating on-the-fly modifications to the image instead of requiring a complete rebuild from scratch. First you need to create a live image with EXT4 as the following:
+
+    `make ROOTFS_FORMAT=ext4 live`
+
+1. To streamline the generation of an up-to-date live image, especially during iterative development cycles, you can patch the EXT4 rootfs partition with generated rootfs tar. This approach minimizes the time required for generating the whole live image. Below is the command for updateing the EXT4 live image generated by the previous command:
+
+    `make LIVE_UPDATE=1 live`
```
Comment (deitch) [inline docs/BUILD.md:390]: This also implies `ROOTFS_FORMAT=ext4`, if I remember the Makefile correctly? We should say it here.

**Review by deitch — CHANGES_REQUESTED** [review]
I like this, nice and elegant. My only comment is on the docs, and very little at that.

**Review by deitch — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -940,12 +962,37 @@ endif
...
+# detect when qcow2 should be updated with a tarball and when it should
+# be recreated from scratch.
+%.qcow2: %.raw $(ROOTFS_TAR) | $(DIST)
+#	Detect if the first %.raw ($<) prerequisite is in the "$?" list,
+#	which means qcow has to be fully recreated. If not - just update
+#	with the existing tar.
+	$(eval RECREATE := $(if $(filter $<,$?),1,0))
+#	Convert raw to qcow or update rootfs with all files from tar:
+#	guestfish one line magic.
+	$(QUIET)if [ "$(RECREATE)" = "1" ]; then \
```
Comment (rene) [inline Makefile:976]: if you are just using RECREATE here, don't you think it's better to just integrate the checking to the if and remove the need of `eval`, like this `[  "$(if .....)" = "1" ]; then...`  ?

**Review by eriknordmark — APPROVED** [review]
Kick off tests

File: Makefile (hunk 3 above)
Comment (rouming) [reply Makefile:976]: I moved it to a separate flag not to overload the if branch and to have a separate description, what exactly the RECREATE flag means. I would keep it as is.
